### PR TITLE
Msgr

### PR DESCRIPTION
**Pull request by live627**
This is a continuation of #174 by @Antes to include upgrade.

### CODE REVIEW

File: other/upgrade_2-1_mysql.sql [hunk 1]
```diff
@@ -196,6 +196,54 @@ CHANGE `stars` `icons` varchar(255) NOT NULL DEFAULT '';
...
+('cust_msn', 'MSN/Live', 'Your Live Messenger email address', 'text', 50, '', 'email', 0, 1, 'forumprofile', 0, 1, 0, 0, '', '<a class="msn" href="http://members.msn.com/{INPUT}" target="_blank" title="Live - {INPUT}"><img src="{IMAGES_URL}/fields/msntalk.gif" alt="Live - {INPUT}"></a>', 1),
+('cust_yahoo', 'Yahoo! Messenger', 'This is your Yahoo! Instant Messenger nickname.', 'text', 50, '', 'email', 0, 1, 'forumprofile', 0, 1, 0, 0, '', '<a class="yim" href="http://edit.yahoo.com/config/send_webmesg?.target={INPUT}" target="_blank" title="Yahoo! Messenger - {INPUT}"><img src="http://opi.yahoo.com/online?m=g&t=0&u={INPUT}" alt="Yahoo! Messenger - {INPUT}"></a>', 1);
+---#
+
+---# Converting member values...
+---{
+// We cannot do this twice
+if (@$modSettings['smfVersion'] < '2.2')
+{
+	// Anyone who can currently post unapproved topics we assume can create drafts as well ...
```
Comment (mikemill) [inline other/upgrade_2-1_mysql.sql]: What does this comment have to do with converting the messenger fields?

File: other/upgrade_2-1_mysql.sql [hunk 2]
```diff
@@ -196,6 +196,54 @@ CHANGE `stars` `icons` varchar(255) NOT NULL DEFAULT '';
...
+			INSERT INTO {$db_prefix}themes
+				(id_member, id_theme, variable, value)
+			VALUES
+				" . implode(',', $inserts));
+}
+---}
+---#
+
+---# Dropping old fields
+ALTER TABLE `s{$db_prefix}members`
```
Comment (mikemill) [inline other/upgrade_2-1_mysql.sql]: Why the s in front of {$db_prefix}?

**Comment by live627** [comment]
> We really store the user's profile data in the theme table? Seems that either that table is named wrong or we are using the wrong table to store that. I know that is OT, but I wanted to make mention.

It's the wrong table. I'm guessing the custom profile fields was a rush job and whomever developed it cut corners by slapping user data in the best matching table. Or was lazy.

Do NOT rename the table because it stores theme-specific variables.

**Comment by jdarwood007** [comment]
The code was pulled from Grudges mod, so I assume it was done because he coded it to be compatible with his mod.  It should be moved to a user_options table.  So should all the other theme options to a theme_options table.

File: other/upgrade_2-1_postgresql.sql [hunk 3]
```diff
@@ -296,12 +350,12 @@ if (@$modSettings['smfVersion'] < '2.1')
 		FROM {$db_prefix}board_permissions
 		WHERE permission = 'post_unapproved_topics'");
 	$inserts = array();
-	while ($row = mysql_fetch_assoc($request))
```
Comment (norv) [inline other/upgrade_2-1_postgresql.sql:299]: This is a good catch, but the fix doesn't belong in this series of commits. Please split the issues of old messenger fields, respectively fixes for drafts support in postgresql, in two different PRs.

File: other/upgrade_2-1_sqlite.sql [hunk 4]
```diff
@@ -238,6 +238,60 @@ upgrade_query("
 ---#
 
 /******************************************************************************/
+--- Messager fields
```
Comment (norv) [inline other/upgrade_2-1_sqlite.sql]: Typo

File: other/upgrade_2-1_postgresql.sql (hunk 3 above)
Comment (emanuele45) [inline other/upgrade_2-1_postgresql.sql:299]: Fixed (along with others) in commit 119eb7f5e19b495566b74e14e0990ab743bfe894
Thanks for noting!

**Comment by XinYenFon** [comment]
isn't it be better if upgrade script asks they want those fields or not.

Quick question (bit off topic) : if i delete my repo & refork, it effects this PR?

**Comment by emanuele45** [comment]
> Quick question (bit off topic) : if i delete my repo & refork, it effects this PR?

Interesting question, but unfortunately I have no answers...
If you are considering to delete the account feel free to drop me a PM or something, I may clone your repo and branches here and keep the changes.

If you wonder why the PR has not yet been integrated, I can tell you why **I** didn't merge it and it is because I prefer to deal with one problem at a time (that is 2.1 at the moment :P).

**Comment by joshuaadickerson** [comment]
@Antes, I think it doesn't matter much if we ask them if they want to keep those fields or not. After they upgrade they can remove them.

**Comment by live627** [comment]
> Quick question (bit off topic) : if i delete my repo & refork, it effects this PR?

Nope! You hungry? Om nom nom tasty repo :P

**Comment by live627** [comment]
@joshuaadickerson I like that way of thinking. Simple.

**Comment by Arantor** [comment]
There are advantages to using the themes table for this - partly because it's the only structure in SMF that provides for multiple arbitrary sized values per user, and partly because when you load a user's theme options, it loads their custom fields for free (e.g. for profiles)

**Comment by XinYenFon** [comment]
@live627 how I missed that I don't know but I missed to sign-off commits can you please close this and take all changes to yourself and sign-off all.

**Comment by matthew-kerle** [comment]
Talked to @Oldiesmann, we decided that we're not going to merge this in SMF 2.1.x as of now. Thanks for your interest and submitting this, I think if we could have gotten it in earlier it would've been a good thing.

**Comment by XinYenFon** [comment]
This is not for 2.1 actually as you can see its for Dev branch more likely for 2.2

**Comment by matthew-kerle** [comment]
The dev branch is in the SMF 2.1 repo, actually.

**Comment by emanuele45** [comment]
Well, the name of the repo was 2.1 for a specific reason (i.e. the 2.x line at the time was considered dead with 2.1 because people were working on 3.0), not because it shouldn't be used for further 2.x version. And if you are planning to continue the 2.x line, this repo should be renamed and used for further development.

**Pull request closed without merging**